File: scratch/test_api.js
async function run() {
  const loginUrl = "https://dev-siteops-platform.transvolt.org/api/v1/auth/login";
  const params = new URLSearchParams();
  params.append("username", "admin");
  params.append("password", "admin123");

  try {
    console.log("Attempting login...");
    const response = await fetch(loginUrl, {
      method: "POST",
      headers: {
        "Content-Type": "application/x-www-form-urlencoded"
      },
      body: params
    });

    console.log("Login Status:", response.status);
    const resData = await response.json();
    
    if (resData && resData.data && resData.data.access_token) {
      const token = resData.data.access_token;
      console.log("Success! Token obtained.");

      // Onboarding counts
      const countsRes = await fetch("https://dev-siteops-platform.transvolt.org/api/v1/onboarding/counts", {
        headers: {
          "Authorization": `Bearer ${token}`
        }
      });
      console.log("Counts Status:", countsRes.status);
      const countsData = await countsRes.json();
      console.log("Counts Data:", JSON.stringify(countsData, null, 2));

      // Recent activities
      const actRes = await fetch("https://dev-siteops-platform.transvolt.org/api/v1/onboarding/recent-activities", {
        headers: {
          "Authorization": `Bearer ${token}`
        }
      });
      console.log("Activities Status:", actRes.status);
      const actData = await actRes.json();
      console.log("Activities Data:", JSON.stringify(actData, null, 2));
    } else {
      console.log("Unexpected response format:", resData);
    }
  } catch (error) {
    console.error("Error:", error);
  }
}

run();

File: scratch/test_api.py
import requests

login_url = "https://dev-siteops-platform.transvolt.org/api/v1/auth/login"
payload = {
    "username": "admin",
    "password": "admin123"
}
headers = {
    "Content-Type": "application/x-www-form-urlencoded"
}

try:
    print("Attempting login...")
    response = requests.post(login_url, data=payload, headers=headers)
    print("Status Code:", response.status_code)
    response_data = response.json()
    print("Response keys:", response_data.keys())
    
    if "data" in response_data and "access_token" in response_data["data"]:
        token = response_data["data"]["access_token"]
        print("Success! Token obtained.")
        
        # Now fetch onboarding counts
        auth_headers = {
            "Authorization": f"Bearer {token}",
            "Content-Type": "application/json"
        }
        
        print("\nFetching onboarding counts...")
        counts_res = requests.get("https://dev-siteops-platform.transvolt.org/api/v1/onboarding/counts", headers=auth_headers)
        print("Counts Status:", counts_res.status_code)
        print("Counts Data:", counts_res.json())
        
        print("\nFetching recent activities...")
        activities_res = requests.get("https://dev-siteops-platform.transvolt.org/api/v1/onboarding/recent-activities", headers=auth_headers)
        print("Activities Status:", activities_res.status_code)
        try:
            print("Activities Data:", activities_res.json())
        except Exception as e:
            print("Failed to parse activities json:", e)
            
    else:
        print("Response data structure:")
        print(response_data)
except Exception as e:
    print("Error:", e)
